### PR TITLE
[Spark-8092][ml] Allow OneVsRest Classifier feature and label column names to be configurable.

### PR DESCRIPTION
**Pull request by harsha2010**
The base classifier input and output columns are ignored in favor of  the ones specified in OneVsRest.

### CODE REVIEW

**Comment by harsha2010** [comment]
@jkbradley please review. i didn't realize PredictorParams had no setters and setLabelCol, etc were in Predictor which OneVsRest does not extend. This means OneVsRest needs to expose setters for label, features etc that the underlying classifier needs.

**Comment by feynmanliang** [comment]
This LGTM. However, after reading this I feel that `OneVsRestModel` should extend `ClassificationModel`. I've created SPARK-8799 for that.

**Comment by jkbradley** [comment]
Sorry for the delay!  Will review now.  Jenkins test this please

**Comment by jkbradley** [comment]
I guess there are merge conflicts...but will comment anyways

File: mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala [hunk 1]
```diff
@@ -192,7 +201,11 @@ final class OneVsRest(override val uid: String)
       val labelUDFWithNewMeta = labelUDF.as(labelColName, newLabelMeta)
       val trainingDataset = multiclassLabeled.withColumn(labelColName, labelUDFWithNewMeta)
       val classifier = getClassifier
-      classifier.fit(trainingDataset, classifier.labelCol -> labelColName)
+      val paramMap = new ParamMap()
+      paramMap.put(classifier.labelCol -> labelColName)
+      paramMap.put(classifier.featuresCol -> getFeaturesCol)
+      paramMap.put(classifier.predictionCol -> getPredictionCol)
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala]: Is this saying that the base classifier should output "predictionCol"?  It probably should send output to a temp col, to be removed.

**Comment by jkbradley** [comment]
@harsha2010   We should document clearly that the input and output columns of the base classifier are ignored.

**Comment by jkbradley** [comment]
Ping : )

**Comment by harsha2010** [comment]
oops, thanks for reminding me...will update the merged patch tonight

**Comment by jkbradley** [comment]
Could you please add doc to the "classifier" Param indicating that the base classifier input and output columns are ignored in favor of the ones specified in OneVsRest?  Other than that, it looks good.

**Comment by jkbradley** [comment]
LGTM  Would you mind adding a description before we merge this, e.g., to say that the OneVsRest input/output col params override the base classifier ones?  Thanks!

**Comment by harsha2010** [comment]
@jkbradley , sure, do you mean in the Spark docs? The description is already in the code docs for the params of OneVsRest.

**Comment by jkbradley** [comment]
Oh, no, I mean the PR description (the first comment), to become part of the commit message.

**Comment by harsha2010** [comment]
@jkbradley , thanks., added the documentation to the first comment in this PR

**Comment by jkbradley** [comment]
Thanks!  I'll merge this with master

**Pull request closed without merging**